### PR TITLE
[PHI] Add rnn and searchsorted output defs

### PR DESCRIPTION
**Pull request by AndPuQing**
### PR types
Bug fixes
### PR changes
OPs

### Describe

- https://github.com/PaddlePaddle/Paddle/issues/51292

### CODE REVIEW

**Comment by luotao1** [comment]
请解决下冲突

**Comment by AndPuQing** [comment]
> 请解决下冲突

done

File: paddle/phi/kernels/cpu/searchsorted_kernel.cc [hunk 1]
```diff
@@ -25,4 +25,6 @@ PD_REGISTER_KERNEL(searchsorted,
                    float,
                    double,
                    int,
-                   int64_t) {}
+                   int64_t) {
+  kernel->OutputAt(0).SetDataType(phi::DataType::UNDEFINED);
```
Comment (From00) [inline paddle/phi/kernels/cpu/searchsorted_kernel.cc:29]: searchsorted还有一个gpu的kernel需要注册

**Review by From00 — APPROVED** [review]
LGTM

**Pull request merged**